### PR TITLE
Add regression test cases

### PR DESCRIPTION
**Pull request by j-emberton**
Added a selection of regression test cases supplied by Gilberto.

Cases selected to generate 'maximum' number of error reports, with minimum actual tests. Evaluation is summarised in screen cap of excel file. Hopefully this avoids us needing to use an external file storage system to host the test case data. Git LFS is a non-starter due to it being a fork of an open repo.

All tests pass.

### CODE REVIEW

**Comment by j-emberton** [comment]
Plan to rebase to master shortly

**Comment by j-emberton** [comment]
Summary of what cases declare what messages in their respective report files

Screenshot 2024-09-10 at 10 34 10

File: tests/regression_tests/test_ustar_cp/test_ustar_cp.py [hunk 1]
```diff
@@ -77,7 +79,9 @@ def test_ustar_cp(testcase, expected_values, setup_folders, matlab_engine, find_
     # Step 6: Assert that the MATLAB function returns the expected exit code
     assert exitcode == expected_values[0], f"Expected exit code {expected_values[0]}, but got {exitcode}."
 
-
+    # Step 7: Assert that the reference output folder and test run output contain the same files
+    comparison = filecmp.dircmp(ref_outputs, test_outputs)
+    assert comparison, f"Expected True, but returned False"
```
Comment (casterkay) [inline tests/regression_tests/test_ustar_cp/test_ustar_cp.py]: Maybe could be more specific, like "The reference output folder and test run output do not contain the same files."

**Comment by dorchard** [comment]
> Summary of what cases declare what messages in their respective report files
> 
> Screenshot 2024-09-10 at 10 34 10

That's great. Can we serialise this in a textual format and put it with the tests? Otherwise I think this is all good to be merged.

**Comment by j-emberton** [comment]
@dorchard have added a txt summary above the test case list. Is this what you had in mind?

File: tests/regression_tests/test_ustar_cp/test_ustar_cp.py [hunk 2]
```diff
@@ -10,10 +10,67 @@
 
 import pytest
 import io
+import filecmp
+
+"""
+  Chosen test cases to giev adequate coverage:
```
Comment (dorchard) [inline tests/regression_tests/test_ustar_cp/test_ustar_cp.py]:
```suggestion
  Chosen test cases (identified by their site code; first 5 characters) to give adequate coverage (with data files in `tests/test_artifacts`)
```

**Pull request merged**